src/ansys/geometry/core/shapes/curves/nurbs.py

```diff
@@ -141,6 +142,48 @@ def from_control_points(
 
         return curve
 
+    @classmethod
+    @check_input_types
+    def fit_curve_from_points(
+        cls,
+        points: list[Point3D],
+        degree: int,
+    ) -> "NURBSCurve":
+        """Fit a NURBS curve to a set of points.
+
+        Parameters
+        ----------
+        points : list[Point3D]
+            Points to fit the curve to.
+        degree : int
+            Degree of the curve.
+
+        Returns
+        -------
+        NURBSCurve
+            Fitted NURBS curve.
+
+        """
+        from geomdl import fitting
+
+        # Convert points to a format suitable for the fitting function
+        converted_points = []
+        for pt in points:
+            pt = (pt.x.m, pt.y.m, pt.z.m)
+            converted_points.append(pt)
+
+        # Fit the curve to the points
+        curve = fitting.interpolate_curve(converted_points, degree)
+
+        # Construct the NURBSCurve object
+        nurbs_curve = cls()
+        nurbs_curve._nurbs_curve.degree = curve.degree
+        nurbs_curve._nurbs_curve.ctrlpts = curve.ctrlpts
```

Comment:
```suggestion
        nurbs_curve._nurbs_curve.ctrlpts = [Point3D(entry) for entry in curve.ctrlpts]
```
Let's store them as Point3D objects just in case... This works as you can see in the follow up to my previous example:

![image](https://github.com/user-attachments/assets/38d1d1b0-a3cc-4d38-a94b-cefbabdba63f)